File: CodeWars/7kyu-Get-The-Middle-Character.js
// First Version
/* 
const getMiddle = (s) => {
  //Code goes here!
  if(s.length > 0 && s.length < 1000){
    const _p = (s.length % 2 === 0)? true: false;
    if(_p){
      const _id = s.length/2 - 1;
      return s[_id] + s[_id+1];
    } else {
      const _id = (s.length-1)/2;
      return s[_id];
    }
  }
};
*/

const getMiddle = (s) => {
    //Code goes here!
    if(s.length > 0 && s.length < 1000){
      const _mid = Math.floor(s.length / 2);
      return (s.length % 2 === 0)? s[_mid - 1] + s[_mid]: s[_mid];
    }
  };

// Sample Tests:
Test.describe("GetMiddle", function() {
    Test.it("Tests", function() {
      Test.assertEquals(getMiddle("test"),"es");
      Test.assertEquals(getMiddle("testing"),"t");
      Test.assertEquals(getMiddle("middle"),"dd");
      Test.assertEquals(getMiddle("A"),"A");
    });
  });

File: CodeWars/Part1/7kyu-Get-The-Middle-Character.rs
use std::str;

fn get_middle(s:&str) -> &str {
    let len: usize = s.len() as usize;
    let from = (len -1) / 2;
    let to = (len / 2) + 1;
    &s[from .. to]
}

// Sample Tests:
#[test]
fn example_tests() {
  assert_eq!(get_middle("test"),"es");
  assert_eq!(get_middle("testing"),"t");
  assert_eq!(get_middle("middle"),"dd");
  assert_eq!(get_middle("A"),"A");
  assert_eq!(get_middle("of"),"of");
}